platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Polygon.java

```diff
@@ -2,18 +2,48 @@
 
 import android.graphics.Color;
 
+import com.mapbox.mapboxsdk.geometry.LatLng;
 import com.mapbox.mapboxsdk.maps.MapboxMap;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Polygon is a geometry annotation that's a closed loop of coordinates.
  */
 public final class Polygon extends MultiPoint {
 
+    private List<List<LatLng>> holes;
     private int fillColor = Color.BLACK; // default fillColor is black
     private int strokeColor = Color.BLACK; // default strokeColor is black
 
     Polygon() {
         super();
+        holes = new ArrayList<>();
+    }
+
+    /**
+     * Returns a copy of the holes.
+     *
+     * @return A {@link List} of holes.
+     */
+    public List<List<LatLng>> getHoles() {
+        return new ArrayList<>(holes);
+    }
+
+    /**
+     * Sets the holes of this polygon. This method will take a copy of the holes, so further
+     * mutations to holes will have no effect on this polygon.
+     *
+     * @param points A {@link List} {@link List}s of {@link LatLng} points making up the holes.
+     */
+    public void setHoles(List<? extends List<LatLng>> holes) {
+        this.holes = new ArrayList<>(holes);
```

Comment:
Can just update the member list with the collection, so the reference is not exposed.